### PR TITLE
[libc++] Update some C++23 statuses to "Nothing to do" or "Complete"

### PR DESCRIPTION
**Pull request by strega-nil**
- [P2160R1][] "Locks lock lockables"
- [P2212R2][] "Relax Requirements for `time_point::clock`"
- [P1675R2][] "`rethrow_exception` must be allowed to copy"
- [P2340R1][] "Clarifying the status of the 'C headers'"
- [P2460R2][] "Relax requirements on `wchar_t` to match existing practices"

Are all papers that change wording without changing implementation behaviour.

Additionally, [P2736R2][] "Referencing The Unicode Standard", is an already complete paper in 19.0 (as of [LLVM-86543][])

[P2160R1]: https://wg21.link/p2160r1
[P2212R2]: https://wg21.link/p2212r2
[P1675R2]: https://wg21.link/p1675r2
[P2340R1]: https://wg21.link/p2340r1
[P2460R2]: https://wg21.link/p2460r2
[P2736R2]: https://wg21.link/p2736r2
[LLVM-86543]: https://github.com/llvm/llvm-project/pull/86543

### CODE REVIEW

File: libcxx/docs/Status/Cxx23Papers.csv [hunk 1]
```diff
@@ -120,4 +120,4 @@
 "`P2614R2 <https://wg21.link/P2614R2>`__","LWG", "Deprecate ``numeric_limits::has_denorm``","February 2023","|Complete|","18.0",""
 "`P2588R3 <https://wg21.link/P2588R3>`__","LWG", "``barrier``’s phase completion guarantees","February 2023","","",""
 "`P2763R1 <https://wg21.link/P2763R1>`__","LWG", "``layout_stride`` static extents default constructor fix","February 2023","","",""
-"`P2736R2 <https://wg21.link/P2736R2>`__","CWG","Referencing The Unicode Standard","February 2023","","","|format|"
+"`P2736R2 <https://wg21.link/P2736R2>`__","CWG","Referencing The Unicode Standard","February 2023","Nothing to do","","|format|"
```
Comment (frederick-vs-ja) [inline libcxx/docs/Status/Cxx23Papers.csv:123]: Not sure whether this should be considered completed in #86543 (and hence should be in 19.0). @mordante

Comment (frederick-vs-ja) [reply libcxx/docs/Status/Cxx23Papers.csv:123]: Hmm, the referenced Unicode version used by `<format>` seemed to be Unicode 14? (per Bibliography in the paper)

Comment (strega-nil) [reply libcxx/docs/Status/Cxx23Papers.csv:123]: This paper, in my opinion, just removes the referenced version - the current C++ standard only refers to the latest unicode standard, unversioned. In my opinion, this means that implementations are free to reference any standard (perhaps only newer than the C++ standard they are implementing), and therefore that updating the standard referenced in implementations is just an implementation detail.

Comment (mordante) [reply libcxx/docs/Status/Cxx23Papers.csv:123]: Actually it means implementations should _always_ use the latest released Unicode version. This means it's complete until the next Unicode release. I think marking it as complete in version 19.0 would be better.

There is a [core issue](https://github.com/cplusplus/papers/issues/1736) filed against this change.

**Review by mordante — APPROVED** [review]
Thanks a lot for reviewing the status of these papers!
LGTM modulo one comment.

File: libcxx/docs/Status/Cxx23Papers.csv (hunk 1 above)
Comment (strega-nil) [reply libcxx/docs/Status/Cxx23Papers.csv:123]: Mmh, point; I'll make this change, thanks @mordante!

**Pull request merged**